src/PeopleSDKAdapter.js

```diff
@@ -184,4 +185,28 @@ export default class PeopleSDKAdapter extends PeopleAdapter {
 
     return this.getPersonObservables[ID];
   }
+
+  /**
+   * Returns an observable that emits a list of people that match the given query.
+   * An empty array is returned when there are no matches.
+   *
+   * @param {string} query Search query
+   * @returns {external:Observable.<Person[]>} Observable that emits person list based on search query
+   */
+  searchPeople(query) {
+    logger.debug('PEOPLE', undefined, 'searchPeople()', ['called with', {query}]);
+
+    return new Observable(async (observer) => {
+      try {
+        const peopleList = await this.datasource.people.list({displayName: query});
```

Comment:
You cannot use an `async` function as an observer, you need to use a different operator to start the observable chain that handles `async` otherwise this might lead to race conditions or other unexpected bugs. For promises that need to run afresh every time, you should use [`defer`](https://rxjs.dev/api/index/function/defer)
```suggestion
        return defer(this.datasource.people.list({displayName: query}));
```